radix-engine-tests/tests/min_resource_system.rs

```diff
@@ -0,0 +1,103 @@
+use radix_engine::types::*;
+use scrypto::prelude::FromPublicKey;
+use scrypto_unit::*;
+use transaction::builder::ManifestBuilder;
+
+enum ExpectedBehavior {
+    Success,
+    Failure(&'static str),
+}
+
+fn test_mini_resource_system(test_case: &str, expected: ExpectedBehavior) {
+    // Basic setup
+    let mut test_runner = TestRunner::builder().build();
+    let (public_key, _, account) = test_runner.new_allocated_account();
+
+    // Publish package
+    let package_address =
+        test_runner.compile_and_publish("./tests/blueprints/mini_resource_system");
+
+    // Run test case
+    let receipt = test_runner.execute_manifest(
+        ManifestBuilder::new()
+            .lock_fee(account, 100.into())
+            .call_function(package_address, "MiniUser", test_case, manifest_args!())
+            .build(),
+        vec![NonFungibleGlobalId::from_public_key(&public_key)],
+    );
+
+    match expected {
+        ExpectedBehavior::Success => {
+            receipt.expect_commit_success();
+        }
+        ExpectedBehavior::Failure(msg) => {
+            let error_message = receipt
+                .expect_commit_failure()
+                .outcome
+                .expect_failure()
+                .to_string();
+            if !error_message.contains(msg) {
+                panic!("Expected {}, but was {}", msg, error_message);
+            }
+        }
+    };
+}
+
+// Note that we may decide to disallow borrowed reference in Scrypto, then
```

Comment:
Can we have these tests which disallow borrowed references in Scrypto so that we can verify they'll fail?

Reply:
When disallowing borrowed reference in `create_object/unlock_substate`, we will have dedicated tests to verify failure and disable these tests.